File: Server/src/RankInfo.cpp
#include "Model/RankInfo.h"

using namespace Server;

RankInfo::RankInfo(String^ name)
{
    PlayerName = name;
    Wins = 0;
    Losses = 0;
    Score = 0;
}

void RankInfo::AddWin()
{
    Wins++;
    Score += 10;
}

void RankInfo::AddLoss()
{
    Losses++;
    Score -= 5;
}

File: Server/Model/RankInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Server.Model
{
    public class RankInfo
    {
        public string PlayerId { get; set; } = "";
        public int Wins { get; set; } = 0;
        public int Losses { get; set; } = 0;
        public int Score { get; set; } = 1000;
    }
}
